spec/models/spree/gateway/amazon_spec.rb

```diff
@@ -426,40 +490,57 @@ def build_mws_capture_response(state:, total:)
-          "RefundDetails" => {
-            "AmazonRefundId" => "P01-1234567-1234567-0000003",
-            "RefundReferenceId" => "test_refund_1",
-            "SellerRefundNote" => "Lorem ipsum",
-            "RefundType" => "SellerInitiated",
-            "RefundedAmount" => {
-              "CurrencyCode" => "USD",
-              "Amount" => total
-            },
-            "FeeRefunded" => {
-              "CurrencyCode" => "USD",
-              "Amount" => "0"
-            },
-            "RefundStatus" => {
-              "State" => state,
-              "LastUpdateTimestamp" => "2012-11-07T19:10:16Z"
-            },
-            "CreationTimestamp" => "2012-11-05T19:10:16Z"
-          }
-        },
-        "ResponseMetadata" => { "RequestId" => "b4ab4bc3-c9ea-44f0-9a3d-67cccef565c6" }
-      }
-    }
+    <<-XML.strip_heredoc
+      <RefundResponse xmlns="https://mws.amazonservices.com/schema/OffAmazonPayments/2013-01-01">
+        <RefundResult>
+          <RefundDetails>
+            <AmazonRefundId>P01-1234567-1234567-0000003</AmazonRefundId>
+            <RefundReferenceId>test_refund_1</RefundReferenceId>
+            <SellerRefundNote></SellerRefundNote>
+            <RefundType>SellerInitiated</RefundType>
+            <RefundedAmount>
+              <CurrencyCode>USD</CurrencyCode>
+              <Amount>#{total}</Amount>
+            </RefundedAmount>
+            <FeeRefunded>
+              <CurrencyCode>USD</CurrencyCode>
+              <Amount>0</Amount>
+            </FeeRefunded>
+            <RefundStatus>
+              <State>#{state}</State>
+              <LastUpdateTimestamp>2012-11-07T19:10:16Z</LastUpdateTimestamp>
+            </RefundStatus>
+            <CreationTimestamp>2012-11-05T19:10:16Z</CreationTimestamp>
+          </RefundDetails>
+        </RefundResult>
+        <ResponseMetadata>
+          <RequestId>b4ab4bc3-c9ea-44f0-9a3d-67cccef565c6</RequestId>
+        </ResponseMetadata>
+      </RefundResponse>
+    XML
   end
 
   def build_mws_void_response
-    {
-      "CancelOrderReferenceResponse" => {
-        "CancelOrderReferenceResult"=> nil,
-        "ResponseMetadata" => { "RequestId" => "b4ab4bc3-c9ea-44f0-9a3d-67cccef565c6" }
-      }
-    }
+    <<-XML.strip_heredoc
+        <CancelOrderReferenceResponse
+  xmlns="https://mws.amazonservices.com/schema/OffAmazonPayments/2013-01-01">
+        <ResponseMetadata>
+          <RequestId>5f20169b-7ab2-11df-bcef-d35615e2b044</RequestId>
+        </ResponseMetadata>
+      </CancelOrderReferenceResponse>
+    XML
+  end
+
+  def build_error_response(code: 400, message: 'FAILURE')
+    <<-XML.strip_heredoc
+      <ErrorResponse xmlns="http://mws.amazonservices.com/schema/OffAmazonPayments/2013-01-01">
+        <Error>
+          <Type>Sender</Type>
+          <Code>#{code}</Code>
+          <Message>#{message}</Message>
+        </Error>
+        <RequestId>7a1a4219-7b80-4d32-a08c-708fd7f52ebc</RequestId>
+      </ErrorResponse>
+    XML
   end
 end
```

Comment:
An alternative to specifying manually the body of foreign requests is to use the excellent [VCR library](https://github.com/vcr/vcr). Eric and I have used it on numerous occasions on a few projects where you can see examples of it's usage. 

Generally, if you are able to connect and receive response with a test account, it will be a great choice.